modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -1346,6 +1210,136 @@ else if ((objectField.getListTypeDefinitionId() != 0) &&
 		return values;
 	}
 
+	private void _upsertAndRelateNestedObjectEntry(
+			DTOConverterContext dtoConverterContext,
+			Map<String, Object> nestedObjectEntryProperties,
+			ObjectRelationship objectRelationship,
+			ObjectDefinition relatedObjectDefinition, boolean reverse,
+			com.liferay.object.model.ObjectEntry serviceBuilderObjectEntry)
+		throws Exception {
+
+		com.liferay.object.model.ObjectEntry newServiceBuilderObjectEntry =
+			_upsertNestedObjectEntry(
+				dtoConverterContext, nestedObjectEntryProperties,
+				relatedObjectDefinition);
+
+		if (reverse) {
+			_objectRelationshipService.addObjectRelationshipMappingTableValues(
+				objectRelationship.getObjectRelationshipId(),
+				newServiceBuilderObjectEntry.getPrimaryKey(),
+				serviceBuilderObjectEntry.getPrimaryKey(),
+				new ServiceContext());
+		}
+		else {
+			_objectRelationshipService.addObjectRelationshipMappingTableValues(
+				objectRelationship.getObjectRelationshipId(),
+				serviceBuilderObjectEntry.getPrimaryKey(),
+				newServiceBuilderObjectEntry.getPrimaryKey(),
+				new ServiceContext());
+		}
+	}
+
+	private void _upsertNestedObjectEntries(
+			DTOConverterContext dtoConverterContext,
+			ObjectDefinition objectDefinition, ObjectEntry objectEntry,
+			Map<String, ObjectRelationship> objectRelationships,
+			com.liferay.object.model.ObjectEntry serviceBuilderObjectEntry)
+		throws Exception {
+
+		Map<String, Object> properties = objectEntry.getProperties();
+
+		for (Map.Entry<String, ObjectRelationship> entry :
+				objectRelationships.entrySet()) {
+
+			ObjectRelationship objectRelationship = objectRelationships.get(
+				entry.getKey());
+
+			Object propertyValue = properties.get(entry.getKey());
+
+			if ((propertyValue instanceof List) &&
+				(StringUtil.equals(
+					objectRelationship.getType(),
+					ObjectRelationshipConstants.TYPE_MANY_TO_MANY) ||
+				 (StringUtil.equals(
+					 objectRelationship.getType(),
+					 ObjectRelationshipConstants.TYPE_ONE_TO_MANY) &&
+				  (objectRelationship.getObjectDefinitionId1() ==
+					  objectDefinition.getObjectDefinitionId())))) {
+
+				ObjectDefinition relatedObjectDefinition =
+					_objectDefinitionLocalService.getObjectDefinition(
+						_getRelatedObjectDefinitionId(
+							objectDefinition, objectRelationship));
+
+				List<LinkedHashMap<String, Object>>
+					nestedObjectEntryPropertiesList =
+						(List<LinkedHashMap<String, Object>>)propertyValue;
+
+				for (Map<String, Object> nestedObjectEntryProperties :
+						nestedObjectEntryPropertiesList) {
+
+					_upsertAndRelateNestedObjectEntry(
+						dtoConverterContext, nestedObjectEntryProperties,
```

Comment:
I think we could here generate the ObjectEntry instead the using the map as we agreed